lib/vault/api/sys/namespace.rb

```diff
@@ -0,0 +1,81 @@
+module Vault
+  class Namespace < Response
+    # @!attribute [r] id
+    #   ID of the namespace
+    #   @return [String]
+    field :id
+
+    # @!attribute [r] path
+    #   Path of the namespace, includes parent paths if nested.
+    #   @return [String]
+    field :path
+  end
+
+  class Sys
+    # List all namespaces in a given scope. Ignores nested namespaces.
+    #
+    # @example
+    #   Vault.sys.namespaces #=> { :foo => #<struct Vault::Namespace id="xxxx1", path="foo/" }
+    #
+    #   @return [Hash<Symbol, Namespace>]
+    #
+    #   NOTE: Due to a bug in Vault, to be fixed soon, this method CAN return a pure JSON string if a scoping namespace is provided.
```

Comment:
It would be helpful if we had an issue number for this. Is it a bug in Vault OSS or Vault Ent? If OSS, let's make a bug and track it, and reference it here. If it's in Ent, then let's change this to "Due to a bug in Vault Enterprise". I think it's a good idea to reference (public) GH issues when mentioning bugs so when someone eventually comes around this code again, they can see if that bug is still relevant.